src/RollbarClientSubmitter.mts

```diff
@@ -251,12 +252,15 @@ class RollbarClientSubmitter {
     const payload = this.buildPayload(...parameters);
 
     // Bail out of reporting if shouldIgnoreOccurrence() returns a truthy value
-    const { apiUrl, shouldIgnoreOccurrence } = this.configuration;
+    const { apiUrl, transform, shouldIgnoreOccurrence } = this.configuration;
     if (shouldIgnoreOccurrence && shouldIgnoreOccurrence(payload, this.configuration)) {
       console.info('[ROLLBAR CLIENT] Ignoring occurrence', payload, this.configuration);
       return;
     }
 
+    // Allows consumers to mutate the payload similarly to official client
+    transform(payload.data, this.configuration);
```

Comment:
The official client appears to only pass in payload not configuration. It seems like configuration could be removed here without affecting the original intent of this PR. If so, although it deviates somewhat from the official client naming, we could rename this to `transformPayload` which more clearly describes its intent. Thoughts?

Reply:
I had that initially, though then I saw `shouldIgnoreOccurence` (which matches `checkIgnore` from the official library) also receives configuration (despite the official `checkIgnore` doesn't), and I thought it makes sense for those two to have parity. (but I don't mind either way, if you think it's better not to send in the configuration works for me, we don't need it)

On the naming, in theory I'd prefer matching the official version as it makes porting to ours easier... but I can see a lot of things already don't match 1:1 so I can replace as any porting will require a bunch of renaming regardless.

Reply:
I think if we combine our ideas we'll end up in a good place. I fear if we leave the name `transform` it will encourage people to use it to modify `this.configuration`. If we rename it to `transformPayload`, it's clear the intent is to do so only for the payload data. It still lets you modify it but it's more clear that's out of scope. I think you including it could be a positive in case someone wants to use the latest configuration data to make a decision about how to transform the payload data, so how about that? `transformPayload(payload.data, this.configuration);`

Reply:
Done in https://github.com/babbel/rollbar-client.js/pull/16/commits/3adfa4f47dda7522a055e5927dc1ead274649bbc